test/functional/feature_utxo_set_hash.py

```diff
@@ -29,39 +34,29 @@ def test_muhash_implementation(self):
         mocktime = node.getblockheader(node.getblockhash(0))['time'] + 1
         node.setmocktime(mocktime)
 
-        # Generate 100 blocks and remove the first since we plan to spend its
-        # coinbase
-        block_hashes = self.generate(wallet, 1) + self.generate(node, 99)
-        blocks = list(map(lambda block: from_hex(CBlock(), node.getblock(block, False)), block_hashes))
-        blocks.pop(0)
-
-        # Create a spending transaction and mine a block which includes it
+        # Generate 100 blocks, create spending transaction and mine another block
+        self.generate(wallet, 1) + self.generate(node, 99)
         txid = wallet.send_self_transfer(from_node=node)['txid']
-        tx_block = self.generateblock(node, output=wallet.get_address(), transactions=[txid])
-        blocks.append(from_hex(CBlock(), node.getblock(tx_block['hash'], False)))
+        self.generateblock(node, output=wallet.get_address(), transactions=[txid])
 
-        # Serialize the outputs that should be in the UTXO set and add them to
-        # a MuHash object
+        # Serialize created/spent transaction outputs in each block and
+        # apply (add/remove) them to the MuHash object accordingly
         muhash = MuHash3072()
 
-        for height, block in enumerate(blocks):
-            # The Genesis block coinbase is not part of the UTXO set and we
-            # spent the first mined block
-            height += 2
-
-            for tx in block.vtx:
-                for n, tx_out in enumerate(tx.vout):
-                    coinbase = 1 if not tx.vin[0].prevout.hash else 0
-
-                    # Skip witness commitment
-                    if (coinbase and n > 0):
-                        continue
-
-                    data = COutPoint(int(tx.rehash(), 16), n).serialize()
-                    data += struct.pack("<i", height * 2 + coinbase)
-                    data += tx_out.serialize()
-
-                    muhash.insert(data)
+        for height in range(1, node.getblockcount() + 1):
+            for tx in node.getblock(blockhash=node.getblockhash(height), verbose=3)['tx']:
```

Comment:
Since the run-time difference is insignificant, I think that would be a nice candidate for a follow-up. Even if this PR doesn't get merged, in general it would be nice to investigate which tests could be sped up by using this batch request approach.

Reply:
FYI: I looked at quite a few/most of the long-running tests and couldn't identify any significant enough improvements because of batching so I'm not going to look further into it for now myself - comment can be marked resolved.